File: app/src/main/java/cn/lsmya/restaurant/adapter/MessageAdapter.java
package cn.lsmya.restaurant.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import cn.lsmya.restaurant.R;
import cn.lsmya.library.base.BaseRecyclerViewAdapter;
import cn.lsmya.restaurant.model.MessageModel;

public class MessageAdapter extends BaseRecyclerViewAdapter<MessageAdapter.ViewHolder> {
    public MessageAdapter(Context context, List list, OnChildViewClickListener onChildViewClickListener) {
        super(context, list, onChildViewClickListener);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);
        MessageModel model = (MessageModel) list.get(position);
        holder.title.setText(model.getTitle());
        holder.content.setText(model.getContent());
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_message, parent, false));
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.itemTitle)
        TextView title;
        @BindView(R.id.itemContent)
        TextView content;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);

        }
    }
}

File: app/src/main/java/cn/lsmya/restaurant/adapter/MessageAdapter.kt
package cn.lsmya.restaurant.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import cn.lsmya.restaurant.R
import cn.lsmya.restaurant.base.BaseRecyclerViewAdapter
import cn.lsmya.restaurant.model.MessageModel
import java.util.ArrayList

class MessageAdapter(context: Context?, list: ArrayList<*>?, onChildViewClickListener: OnChildViewClickListener?) : BaseRecyclerViewAdapter<MessageAdapter.ViewHolder>(context, list, onChildViewClickListener) {

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)
        var model = list[position] as MessageModel
        holder.title.text = model.title
        holder.content.text = model.content
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.fragment_list, parent, false))
    }

    class ViewHolder : RecyclerView.ViewHolder {

        lateinit var title: TextView
        lateinit var content: TextView

        constructor(itemView: View) : super(itemView) {
            title = itemView.findViewById(R.id.itemTitle)
            content = itemView.findViewById(R.id.itemTitle)
        }
    }
}